lib/FunctionIsolation/IsolateFunctions.cpp

```diff
@@ -491,18 +501,43 @@ bool IFI::handleIndirectBoundary(const std::vector<Boundary> &Boundaries,
       Builder.CreateBr(ClonedBlocks.returnBlock());
       break;
 
-    case BrokenReturn:
-      throwException(Builder, "A broken return was taken");
-      break;
-    case LongJmp:
-      throwException(Builder, "A longjmp was taken");
-      break;
-    case Killer:
-      throwException(Builder, "A killer block has been reached");
-      break;
-    case Unreachable:
-      throwException(Builder, "An unrechable instruction has been reached");
-      break;
+    case BrokenReturn: {
+      BasicBlock::iterator InsertPoint = Builder.GetInsertPoint();
+      revng_assert(not Builder.GetInsertBlock()->empty());
+      Instruction *Old = InsertPoint == Builder.GetInsertBlock()->end() ?
+                           &*Builder.GetInsertBlock()->rbegin() :
+                           &*InsertPoint;
+      throwException(Builder, "A broken return was taken", Old->getDebugLoc());
+    } break;
+    case LongJmp: {
+      BasicBlock::iterator InsertPoint = Builder.GetInsertPoint();
+      revng_assert(not Builder.GetInsertBlock()->empty());
+      Instruction *Old = InsertPoint == Builder.GetInsertBlock()->end() ?
+                           &*Builder.GetInsertBlock()->rbegin() :
+                           &*InsertPoint;
+      throwException(Builder, "A longjmp was taken", Old->getDebugLoc());
+    } break;
+    case Killer: {
+      BasicBlock::iterator InsertPoint = Builder.GetInsertPoint();
+      revng_assert(not Builder.GetInsertBlock()->empty());
+      Instruction *Old = InsertPoint == Builder.GetInsertBlock()->end() ?
+                           &*Builder.GetInsertBlock()->rbegin() :
+                           &*InsertPoint;
+      throwException(Builder,
+                     "A killer block has been reached",
+                     Old->getDebugLoc());
+    } break;
+    case Unreachable: {
+      BasicBlock::iterator InsertPoint = Builder.GetInsertPoint();
+      revng_assert(not Builder.GetInsertBlock()->empty());
+      Instruction *Old = InsertPoint == Builder.GetInsertBlock()->end() ?
+                           &*Builder.GetInsertBlock()->rbegin() :
+                           &*InsertPoint;
+      throwException(Builder,
+                     "An unrechable instruction has been "
+                     "reached",
+                     Old->getDebugLoc());
+    } break;
```

Comment:
Too much code repetition for something of little relevance.
Why don't you push the boring code into `throwException` and add an argument for the `Instruction` to copy from?

Reply:
I'll take a look into this.

Reply:
Done. Let me know what you think.